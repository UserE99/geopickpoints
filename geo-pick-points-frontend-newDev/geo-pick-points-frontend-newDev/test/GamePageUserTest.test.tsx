import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter, useNavigate, useLocation } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import GamePageUser from '../src/Pages/GamePageUser';
import '@testing-library/jest-dom';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

const mockStore = configureStore([]);

test('renders the GamePageUser component and matches the snapshot', () => {
  const mockNavigate = jest.fn();
  (useNavigate as jest.Mock).mockReturnValue(mockNavigate);

  const store = mockStore({
    players: {
      players: [{ id: '1234', name: 'TestPlayer' }],
      loading: false,
      error: null,
    },
    authentication: {
      authen: {
        isLoggedIn: true,
      },
    },
  });

  (useLocation as jest.Mock).mockReturnValue({
    state: {
      nickName: 'TestPlayer',
      host: true,
      idOfUser: '1',
      email: 'test@example.com',
    },
  });

  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <GamePageUser nickName="TestPlayer" host={true} idOfUser="1" email="test@example.com" />
      </MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});

test('displays the welcome message correctly', () => {
  const mockNavigate = jest.fn();
  (useNavigate as jest.Mock).mockReturnValue(mockNavigate);

  const store = mockStore({
    players: {
      players: [{ id: '1234', name: 'TestPlayer' }],
      loading: false,
      error: null,
    },
    authentication: {
      authen: {
        isLoggedIn: true,
      },
    },
  });

  (useLocation as jest.Mock).mockReturnValue({
    state: {
      nickName: 'TestPlayer',
      host: true,
      idOfUser: '1',
      email: 'test@example.com',
    },
  });

  render(
    <Provider store={store}>
      <MemoryRouter>
        <GamePageUser nickName="TestPlayer" host={true} idOfUser="1" email="test@example.com" />
      </MemoryRouter>
    </Provider>
  );

  const headingElements = screen.getAllByText((content, element) => {
    return element?.textContent?.trim() === "Hallo, TestPlayer !";
  });
  
  expect(headingElements[0]).toBeInTheDocument();});


test('shows HostBuildDialog when HostBuildButton is clicked', () => {
  const mockNavigate = jest.fn();
  (useNavigate as jest.Mock).mockReturnValue(mockNavigate);

  const store = mockStore({
    players: {
      players: [{ id: '1234', name: 'TestPlayer' }],
      loading: false,
      error: null,
    },
    authentication: {
      authen: {
        isLoggedIn: true,
      },
    },
  });

  (useLocation as jest.Mock).mockReturnValue({
    state: {
      nickName: 'TestPlayer',
      host: true,
      idOfUser: '1',
      email: 'test@example.com',
    },
  });

  render(
    <Provider store={store}>
      <MemoryRouter>
        <GamePageUser nickName="TestPlayer" host={true} idOfUser="1" email="test@example.com" />
      </MemoryRouter>
    </Provider>
  );

  fireEvent.click(screen.getByRole('button', { name: /Create Teams/i }));

  expect(screen.getByText(/Create Game to be Host/i)).toBeInTheDocument();
});
