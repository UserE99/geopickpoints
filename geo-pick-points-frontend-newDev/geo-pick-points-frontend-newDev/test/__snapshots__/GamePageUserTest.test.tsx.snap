// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the GamePageUser component and matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-sm-flex align-items-center justify-content-between mb-4"
  >
    <h1
      class="h3 mb-0 text-gray-800"
      style="margin-top: 1.5rem; margin-left: 1.0rem;"
    >
       Hallo, TestPlayer !
    </h1>
  </div>
  <div
    class="row"
    style="margin-left: 0.5rem; margin-right: 0.5rem;"
  >
    <div
      class="col-xl-3 col-md-6 mb-4"
    >
      <div
        class="card border-left-warning shadow h-100 py-2"
        style="cursor: pointer;"
      >
        <div
          class="card-body"
        >
          <div
            class="row no-gutters align-items-center"
          >
            <div
              class="col mr-2"
            >
              <div
                class="text-xs font-weight-bold text-warning text-uppercase mb-1"
              >
                Dashboard
              </div>
              <div
                class="h5 mb-0 font-weight-bold text-gray-800"
              >
                Dashboard
              </div>
            </div>
            <div
              class="col-auto"
            >
              <i
                class="fas fa-comments fa-2x text-gray-300"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-xl-3 col-md-6 mb-4"
    >
      <div
        class="card border-left-primary shadow h-100 py-2"
        style="cursor: pointer;"
      >
        <div
          class="card-body"
        >
          <div
            class="row no-gutters align-items-center"
          >
            <div
              class="col mr-2"
            >
              <div
                class="text-xs font-weight-bold text-primary text-uppercase mb-1"
              >
                Your Profile (Edit)
              </div>
              <div
                class="h5 mb-0 font-weight-bold text-gray-800"
              >
                Name: TestPlayer
              </div>
            </div>
            <div
              class="col-auto"
            >
              <i
                class="[object Object] fa-2x text-gray-300"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
    style="margin-left: 0.5rem;"
  >
    <div
      class=" col-lg-7"
    >
      <div
        class="card shadow mb-4"
      >
        <div
          class="card-header py-3 d-flex flex-row align-items-center justify-content-between"
        >
          <h6
            class="m-0 font-weight-bold text-primary"
          >
            Join Team to play
          </h6>
        </div>
        <div
          class="card-body"
        >
          <div
            class="container"
          >
            <div
              class="vh-20 d-flex justify-content-center align-items-center row"
            >
              <div
                class="col-lg-8 col-md-10 col-12"
              >
                <form>
                  <div
                    class="row"
                  >
                    <div
                      class="text-center text-primary"
                    >
                      Invite-Code:
                    </div>
                    <div
                      class="form-outline"
                    >
                      <input
                        class="form-control form-control-lg"
                        id="code_Test"
                        name="codeInvite"
                        placeholder="enter your code"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="mt-4 "
                  >
                    <button
                      class="btn btn-warning"
                      id="TestID"
                      type="submit"
                    >
                      Go to Lobby Game
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="col-xl-4 col-lg-5"
    >
      <div
        class="card shadow mb-4"
      >
        <div
          class="card-header py-3 d-flex flex-row align-items-center justify-content-between"
        >
          <h6
            class="m-0 font-weight-bold text-primary"
          >
            Create Game to be Host
          </h6>
        </div>
        <div
          class="card-body"
        >
          <button
            class="btn btn-primary"
            data-testid="HostBuildButton"
            type="button"
          >
            Create Teams
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
